app/src/types/Cup.tsx

```diff
@@ -1,6 +1,10 @@
 export default interface Cup {
-    isPublished: Boolean;
-    isOpen: Boolean;
+    isPublished: boolean;
+    isOpen: boolean;
     timeCreated: Date;
     name: string;
 }
+
+export interface CupWithId extends Cup {
```

Comment:
Instead of creating a new interfact can you use the pattern `Entity<Cup>` when `CupWithId` is required?

Reply:
done